pkg/controller/v1alpha1/inferencegraph/controller.go

```diff
@@ -173,9 +178,53 @@ func (r *InferenceGraphReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return reconcile.Result{}, errors.Wrapf(err, "fails to create DeployConfig")
 	}
 
+	// name of our custom finalizer
+	finalizerName := "inferencegraph.finalizers"
+
+	// examine DeletionTimestamp to determine if object is under deletion
+	if graph.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer.
+		if !utils.Includes(graph.ObjectMeta.Finalizers, finalizerName) {
+			graph.ObjectMeta.Finalizers = append(graph.ObjectMeta.Finalizers, finalizerName)
+			patchYaml := "metadata:\n  finalizers: [" + strings.Join(graph.ObjectMeta.Finalizers, ",") + "]"
+			patchJson, _ := yaml.YAMLToJSON([]byte(patchYaml))
+			if err = r.Patch(ctx, graph, client.RawPatch(types.MergePatchType, patchJson)); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted
+		if utils.Includes(graph.ObjectMeta.Finalizers, finalizerName) {
+			// our finalizer is present, so lets handle any external dependency
+			if err = r.onDeleteCleanup(ctx, graph); err != nil {
```

Comment:
I am confused here, if yo are deleting the IG, why are you patching it below?

Reply:
Well... I am not deleting it; the user is deleting it :-) 

Read about the finalizers pattern: https://book.kubebuilder.io/reference/using-finalizers

Reply:
Yes, my point is, why patch it manually?
Like the ISVC, I would expect to create and delete a IG without a need to manually add a finalizer here.

I might be missing something here..

Reply:
I still don't understand the question... I'll answer with the following which I'm not sure is the answer you may be expecting:

Notice that `r` is not the kube client, nor `onDeleteCleanup` is a func to delete the IG. That func is defined below and is cleaning child resources.

Reply:
yeah, but, why are you doing it?
I don't remember seeing anything similar on isvcs. I am trying to understand why forcing the addition of the finalizer is required for IG.

Reply:
If we don't use a finalizer, how can we ensure clean up of the ClusterRoleBinding?

Reply:
I would expect these roles being added when the KServe is installed, like ISVC.

Reply:
Yes, the ClusterRoles this is using are system roles. Those will exist on a typical OpenShift cluster and we don't need to create them.

The clean-up in this part is dealing with the `ClusterRoleBinding`. I don't think we can pre-create the bindings, if we don't know what are the namespaces nor the ServiceAccounts that we should bind privileges to.